service/worker/scanner/executions/task.go

```diff
@@ -177,6 +193,35 @@ func (t *task) getPaginationFn() collection.PaginationFn[*persistencespb.Workflo
 	}
 }
 
+func (t *task) handleFailures(
+	mutableState *MutableState,
+	results []MutableStateValidationResult,
+) error {
+	for _, failure := range results {
+		switch failure.failureType {
+		case mutableStateRetentionFailureType:
+			executionInfo := mutableState.GetExecutionInfo()
+			runID := mutableState.GetExecutionState().GetRunId()
+			_, err := t.historyClient.DeleteWorkflowExecution(t.ctx, &historyservice.DeleteWorkflowExecutionRequest{
+				NamespaceId: executionInfo.GetNamespaceId(),
+				WorkflowExecution: &commonpb.WorkflowExecution{
+					WorkflowId: executionInfo.GetWorkflowId(),
+					RunId:      runID,
+				},
+				WorkflowVersion:    common.EmptyVersion,
+				ClosedWorkflowOnly: true,
+			})
+			if err != nil {
+				return err
```

Comment:
Do we need to handle non-retryable errors like not found or namespace not found here?

Reply:
Yes. Handled not found errors